src/Core/FakeVimCommand.cpp

```diff
@@ -0,0 +1,307 @@
+            return {true, ""};
+        }
+        else if (!file.isFile())
+        {
+
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(tr("Error: %1 is not a file").arg(ex.args),
+                                                                  STATUS_RESPONSE_TIMEOUT);
+            return {true, ""};
+        }
+
+        else if (!Util::cppSuffix.contains(file.suffix()) && !Util::javaSuffix.contains(file.suffix()) &&
+                 !Util::javaSuffix.contains(file.suffix()))
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(
+                    tr("Error: %1 is not a source file of C/C++, Java or Python").arg(ex.args),
+                    STATUS_RESPONSE_TIMEOUT);
+
+            return {true, ""};
+        }
+        else
+        {
+            return {true, ex.args};
+        }
+    }
+    return {false, ""};
+}
+
+bool FakeVimCommand::wantCompile(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "compile" || ex.cmd == "cmp";
+}
+
+bool FakeVimCommand::wantCompileRun(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "crun" || ex.cmd == "crn";
+}
+
+bool FakeVimCommand::wantDetachedRun(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "drun" || ex.cmd == "drn";
+}
+
+QPair<bool, int> FakeVimCommand::wantRun(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "run")
+    {
+        if (ex.args.isEmpty()) // run all testcases
+        {
+            return {true, 0};
+        }
+        bool ok = false;
+        int caseNum = ex.args.toInt(&ok);
+        if (ok)
+            return {true, caseNum};
+        else
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(tr("%1 is not a number").arg(caseNum),
+                                                                  STATUS_RESPONSE_TIMEOUT);
+            return {true, -1};
+        }
+    }
+
+    return {false, -1};
+}
+bool FakeVimCommand::wantKillAll(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "killall" || ex.cmd == "kap"; // kill all process
+}
+bool FakeVimCommand::wantFormat(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "format" || ex.cmd == "fmt";
+}
+bool FakeVimCommand::wantSnippets(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "snippet" || ex.cmd == "snp";
+}
+QPair<bool, QString> FakeVimCommand::wantVModeChange(FakeVim::Internal::ExCommand const &ex)
+{
+
+    if (ex.cmd == "vmode" || ex.cmd == "vmd")
+    {
+        if (ex.args == "split" || ex.args == "edit")
+            return {true, ex.args};
+        else
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(
+                    tr("%1 is not a valid view mode. It should be one of \"split\", \"edit\"").arg(ex.args),
```

Comment:
User cannot switch to IO mode because, in IO mode the editor is hidden and hence vim commands cannot be issued.